dv/uvm/core_ibex/sim.py

```diff
@@ -190,6 +193,74 @@ def get_simulator_cmd(simulator, yaml_path, enables):
             subst_cmd(entry['sim']['cmd'], enables, entry['sim'], env_vars))
 
 
+def filter_tests_by_config(cfg, test_list):
+    '''Filter the provided testlist to ensure all required RTL parameters are
+       provided by the Ibex core configuration.
+
+    Arguments:
+
+        cfg: string name of the ibex config being tested, should match a
+             config name from ibex_configs.yaml.
+
+        test_list: list of test entry objects parsed from the YAML testlist
+
+    Returns:
+
+        filtered_test_list: a list of test entry objects, filtered such that
+                            all tests incompatible with the specified ibex
+                            config have been removed.
+
+                            e.g. if the "small" config has been specified, this
+                            function will filter out all tests that require
+                            B-extension and PMP parameters
+    '''
+    filtered_test_list = []
+    config = parse_config(cfg, os.path.join(_IBEX_ROOT, "ibex_configs.yaml"))
+
+    for test in test_list:
+        if "rtl_params" in test:
+            param_dict = test['rtl_params']
+            matching_params = False
+            for p in param_dict.keys():
+                # Throw an error if required RTL parameters in the testlist
+                # have been formatted incorrectly (typos, wrong parameters, etc)
+                if p not in config:
+                    logging.error('parameter % not found in config %' % (p, cfg))
+                    sys.exit(RET_FAIL)
+                else:
+                    config_val = config.get(p)
+                    p_val = param_dict.get(p)
+
+                    # Ibex has some enum parameters, so as a result some tests
+                    # are able to run with several of these parameter values
+                    # (like bitmanipulation tests).
+                    # If this is the case, the testlist will specify all legal
+                    # enum values, check if any of them match the config.
+                    if isinstance(p_val, list):
+                        for val in p_val:
+                            if val == config_val:
+                                matching_params = True
+                                break;
```

Comment:
Couldn't this be `matching_params = config_val in p_val`?

Reply:
good point! this would remove a lot of the unnecessary indents here, probably not as much need to move the `matching_params` stuff into a separate function now